# A Third-party Analysis of the AASL Standards

Jinren Yuan

This section contains the analysis of **administrators**.

In [1]:
import pandas as pd

In [2]:
df_a = pd.read_csv("data/AASL Standards Survey - Administrators - Sheet.csv")
df_a.shape

(35, 174)

In [3]:
df_a.head()

Respondent ID  Collector ID           Start Date             End Date  \
0            NaN           NaN                  NaN                  NaN   
1   1.144687e+11   428602768.0  2023-11-20 20:38:23  2023-11-20 20:44:20   
2   1.144669e+11   428602768.0  2023-11-17 19:34:16  2023-11-17 19:41:20   
3   1.144663e+11   428602768.0  2023-11-17 07:59:11  2023-11-17 08:04:47   
4   1.144663e+11   428602768.0  2023-11-17 07:29:23  2023-11-17 08:01:35   

   IP Address  Email Address  First Name  Last Name  Custom Data 1  \
0         NaN            NaN         NaN        NaN            NaN   
1         NaN            NaN         NaN        NaN            NaN   
2         NaN            NaN         NaN        NaN            NaN   
3         NaN            NaN         NaN        NaN            NaN   
4         NaN            NaN         NaN        NaN            NaN   

  Which of the following positions most closely reflects your job role? Please select all that apply.  \
0                                  None of the above                                                    
1                                                NaN                                                    
2                                                NaN                                                    
3                                                NaN                                                    
4                                                NaN                                                    

   ...              Unnamed: 164                       Unnamed: 165  \
0  ...  English language version  Another language (please specify)   
1  ...                       NaN                                NaN   
2  ...                       NaN                                NaN   
3  ...                       NaN                                NaN   
4  ...                       NaN                                NaN   

  What is the primary language spoken in your work environment?  \
0                                           Response              
1                                                NaN              
2                                                NaN              
3                                                NaN              
4                                                NaN              

             Unnamed: 167  \
0  Other (please specify)   
1                     NaN   
2                     NaN   
3                     NaN   
4                     NaN   

  Are you of Hispanic or Latino background – such as Mexican, Puerto Rican, Cuban, or another Latin American background?  \
0                                           Response                                                                       
1                                                NaN                                                                       
2                                                NaN                                                                       
3                                                NaN                                                                       
4                                                NaN                                                                       

  Which of the following categories best describes your race or ethnicity?  \
0                                           Response                         
1                                                NaN                         
2                                                NaN                         
3                                                NaN                         
4                                                NaN                         

             Unnamed: 170  \
0  Other (please specify)   
1                     NaN   
2                     NaN   
3                     NaN   
4                     NaN   

  Which of the following categories best describes your employment status?  \
0                     

In [4]:
df_a.columns

Index(['Respondent ID', 'Collector ID', 'Start Date', 'End Date', 'IP Address',
       'Email Address', 'First Name', 'Last Name', 'Custom Data 1',
       'Which of the following positions most closely reflects your job role? Please select all that apply.',
       ...
       'Unnamed: 164', 'Unnamed: 165',
       'What is the primary language spoken in your work environment?',
       'Unnamed: 167',
       'Are you of Hispanic or Latino background – such as Mexican, Puerto Rican, Cuban, or another Latin American background?',
       'Which of the following categories best describes your race or ethnicity?',
       'Unnamed: 170',
       'Which of the following categories best describes your employment status?',
       'Unnamed: 172', 'Is there anything else you would like to add?'],
      dtype='object', length=174)

In [5]:
# drop irrelavent ID and date fields
df_a = df_a.drop(columns=['Respondent ID', 'Collector ID', 'Start Date', 'End Date'])

# drop fields that contain only missing values (NaN)
df_a = df_a.dropna(axis=1, how='all')
df_a.head()

Which of the following positions most closely reflects your job role? Please select all that apply.  \
0                                  None of the above                                                    
1                                                NaN                                                    
2                                                NaN                                                    
3                                                NaN                                                    
4                                                NaN                                                    

                    Unnamed: 10  \
0  Building Level Administrator   
1                           NaN   
2                           NaN   
3                           NaN   
4                           NaN   

                                         Unnamed: 11  \
0  District Level Technology (e.g. tech coordinat...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                    Unnamed: 12     Unnamed: 13          Unnamed: 14  \
0  District Level Administrator  Superintendent  School Board Member   
1  District Level Administrator             NaN                  NaN   
2  District Level Administrator             NaN                  NaN   
3  District Level Administrator             NaN                  NaN   
4  District Level Administrator             NaN                  NaN   

  Which of the following best describes your work environment? Please select all that apply. (i.e. If you are in a combined setting you may select multiple levels.)  \
0                                  Elementary school                                                                                                                   
1                                                NaN                                                                                                                   
2                                                NaN                                                                                                                   
3                                  Elementary school                                                                                                                   
4                                                NaN                                                                                                                   

                           Unnamed: 16  Unnamed: 17         Unnamed: 18  ...  \
0  Middle school or junior high school  High school  Multiple buildings  ...   
1                                  NaN          NaN                 NaN  ...   
2                                  NaN          NaN                 NaN  ...   
3  Middle school or junior high school  High school                 NaN  ...   
4                                  NaN          NaN                 NaN  ...   

               Unnamed: 164                       Unnamed: 165  \
0  English language version  Another language (please specify)   
1                       NaN                                NaN   
2                       NaN                                NaN   
3                       NaN                                NaN   
4                       NaN                                NaN   

  What is the primary language spoken in your work environment?  \
0                                           Response              
1                                                NaN              
2                                                NaN              
3                                                NaN              
4                                                NaN              

             Unnamed: 167  \
0  Other (please specify)   
1                     NaN   
2            

In [6]:
# Drop Q2 - Q4
df_a = df_a.drop(df_a.columns[6:37], axis=1)

In [7]:
df_a.shape

(35, 134)

In [8]:
# Drop from Q32
df_a = df_a.drop(df_a.columns[117:], axis=1)

In [9]:
for col in df_a.columns:
    print(col)

Which of the following positions most closely reflects your job role? Please select all that apply.
Unnamed: 10
Unnamed: 11
Unnamed: 12
Unnamed: 13
Unnamed: 14
What does school library staffing look like in your place of work or the school(s) you support? Please select all that apply.
Unnamed: 47
Unnamed: 48
Unnamed: 49
Unnamed: 50
Unnamed: 51
How often do you collaborate and work with the school librarian(s)?
What are the main reasons you do not work with or collaborate more often with the school librarian?
Who usually initiates collaborations between yourself and the school librarian?
Unnamed: 55
In your opinion, what are the school librarian’s most important contributions to the education program of the school?
Which, if any, of the following leadership opportunities in your organization are school librarians involved? Please select all that apply.
Unnamed: 58
Unnamed: 59
Unnamed: 60
Unnamed: 61
Unnamed: 62
Unnamed: 63
Unnamed: 64
Unnamed: 65
Unnamed: 66
Here is an image of the Nati

In [10]:
df_a.shape

(35, 117)

In [11]:
df_a.head()

Which of the following positions most closely reflects your job role? Please select all that apply.  \
0                                  None of the above                                                    
1                                                NaN                                                    
2                                                NaN                                                    
3                                                NaN                                                    
4                                                NaN                                                    

                    Unnamed: 10  \
0  Building Level Administrator   
1                           NaN   
2                           NaN   
3                           NaN   
4                           NaN   

                                         Unnamed: 11  \
0  District Level Technology (e.g. tech coordinat...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                    Unnamed: 12     Unnamed: 13          Unnamed: 14  \
0  District Level Administrator  Superintendent  School Board Member   
1  District Level Administrator             NaN                  NaN   
2  District Level Administrator             NaN                  NaN   
3  District Level Administrator             NaN                  NaN   
4  District Level Administrator             NaN                  NaN   

  What does school library staffing look like in your place of work or the school(s) you support? Please select all that apply.  \
0                                  None of the above                                                                              
1                                                NaN                                                                              
2                                                NaN                                                                              
3                                                NaN                                                                              
4                                                NaN                                                                              

                             Unnamed: 47  \
0  Certified / licensed school librarian   
1  Certified / licensed school librarian   
2  Certified / licensed school librarian   
3  Certified / licensed school librarian   
4  Certified / licensed school librarian   

                                         Unnamed: 48  \
0  Paraprofessional or other non-instructional staff   
1  Paraprofessional or other non-instructional staff   
2                                                NaN   
3  Paraprofessional or other non-instructional staff   
4                                                NaN   

                     Unnamed: 49  ...  \
0  Teacher on special assignment  ...   
1                            NaN  ...   
2                            NaN  ...   
3                            NaN  ...   
4                            NaN  ...   

                                        Unnamed: 147  \
0  Curate Make meaning for oneself and others by ...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                        Unnamed: 148  \
0  Explore Discover and innovate in a growth mind...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                        Unnamed: 149  \
0  Engage Demonstrate sa

In [12]:
df_a.rename(columns={df_a.columns[0]: 'q1_none', # Q1
                     df_a.columns[1]: 'q1_building',
                     df_a.columns[2]: 'q1_district_technology',
                     df_a.columns[3]: 'q1_district_administrator',
                     df_a.columns[4]: 'q1_superintendent',
                     df_a.columns[5]: 'q1_school_board',

                     df_a.columns[6]: 'q5_none', # Q5
                     df_a.columns[7]: 'q5_certified',
                     df_a.columns[8]: 'q5_noninstructional',
                     df_a.columns[9]: 'q5_special_teacher',
                     df_a.columns[10]: 'q5_volunteer',
                     df_a.columns[11]: 'q5_not_sure',

                     df_a.columns[12]: 'q6_how_often', # Q6
                     df_a.columns[13]: 'q7_main_reason', # Q7

                     df_a.columns[14]: 'q8_initiate', # Q8
                     df_a.columns[15]: 'q8_others',

                     df_a.columns[16]: 'q9_librarians_contribution', # Q9

                     df_a.columns[17]: 'q10_none', # Q10
                     df_a.columns[18]: 'q10_building_leadership',
                     df_a.columns[19]: 'q10_school_improvement',
                     df_a.columns[20]: 'q10_district_leadership',
                     df_a.columns[21]: 'q10_district_improvement',
                     df_a.columns[22]: 'q10_district_strategy',
                     df_a.columns[23]: 'q10_technology_committee',
                     df_a.columns[24]: 'q10_professional_learning',
                     df_a.columns[25]: 'q10_curriculum_committee',
                     df_a.columns[26]: 'q10_others'}
                     , inplace=True)

In [13]:
df_a.rename(columns={df_a.columns[27]: 'q11_no', # Q11
                     df_a.columns[28]: 'q11_yes_purchase',
                     df_a.columns[29]: 'q11_yes_from_workplace',
                     df_a.columns[30]: 'q11_yes_from_librarian',
                     df_a.columns[31]: 'q11_yes_not_own',
                     df_a.columns[32]: 'q11_others',

                     df_a.columns[33]: 'q12_no', # Q12
                     df_a.columns[34]: 'q12_yes_purchase',
                     df_a.columns[35]: 'q12_yes_from_workplace',
                     df_a.columns[36]: 'q12_yes_from_librarian',
                     df_a.columns[37]: 'q12_yes_not_own',
                     df_a.columns[38]: 'q12_others',

                     df_a.columns[39]: 'q13_provide_copy', # Q13
                     df_a.columns[40]: 'q13_others',

                     df_a.columns[41]: 'q14_how_familiar', # Q14
                     df_a.columns[42]: 'q15_rate', # Q15
                     df_a.columns[43]: 'q16_poor_rating_reason', # Q16

                     df_a.columns[44]: 'q17_valuable', # Q17
                     df_a.columns[45]: 'q17_relevant',
                     df_a.columns[46]: 'q17_practical',
                     df_a.columns[47]: 'q17_useful',
                     df_a.columns[48]: 'q17_easy_to_use',
                     df_a.columns[49]: 'q17_easy_to_understand',
                     df_a.columns[50]: 'q17_well_written',
                     df_a.columns[51]: 'q17_nicely_designed',
                     df_a.columns[52]: 'q17_well_organized',
                     df_a.columns[53]: 'q17_ADA_accessible',

                     df_a.columns[54]: 'q18_mirror_other_standards.', # Q18
                     df_a.columns[55]: 'q18_crosswalk_with_other_standards',
                     df_a.columns[56]: 'q18_align_educational_outlook'}
                     , inplace=True)

In [14]:
df_a.rename(columns={df_a.columns[57]: 'q19_unique_and_essential', # Q19
                     df_a.columns[58]: 'q19_effective',
                     df_a.columns[59]: 'q19_prepared',
                     df_a.columns[60]: 'q19_reading',
                     df_a.columns[61]: 'q19_intellectual_freedom',
                     df_a.columns[62]: 'q19_equitably_available',

                     df_a.columns[63]: 'q20_librarian_using', # Q20

                     df_a.columns[64]: 'q21_none', # Q21
                     df_a.columns[65]: 'q21_lesson_preparation_and_assessment',
                     df_a.columns[66]: 'q21_evaluate_librarians_or_libraries',
                     df_a.columns[67]: 'q21_learning_opportunities_for_educators',
                     df_a.columns[68]: 'q21_collaboration_with_teacher',
                     df_a.columns[69]: 'q21_with_other_educators',
                     df_a.columns[70]: 'q21_with_external_stakeholders',
                     df_a.columns[71]: 'q21_with_parents',
                     df_a.columns[72]: 'q21_demonstrate_learner_success',
                     df_a.columns[73]: 'q21_guide_instructional_priorities',
                     df_a.columns[74]: 'q21_develop_curriculum',
                     df_a.columns[75]: 'q21_advocate_for_school_library',
                     df_a.columns[76]: 'q21_advocate_for_learners',
                     df_a.columns[77]: 'q21_increase_understanding_of_school_librarian',
                     df_a.columns[78]: 'q21_support_school_mission',
                     df_a.columns[79]: 'q21_others',

                     df_a.columns[80]: 'q22_direct_instruction', # Q22
                     df_a.columns[81]: 'q22_lesson_development',
                     df_a.columns[82]: 'q22_collaboration',
                     df_a.columns[83]: 'q22_collection',
                     df_a.columns[84]: 'q22_leadership',
                     df_a.columns[85]: 'q22_advocacy',
                     df_a.columns[86]: 'q22_goal_and_planning'}
                     , inplace=True)

In [15]:
df_a.rename(columns={df_a.columns[87]: 'q23_none', # Q23
                     df_a.columns[88]: 'q23_professional_associations',
                     df_a.columns[89]: 'q23_conferences',
                     df_a.columns[90]: 'q23_workshops',
                     df_a.columns[91]: 'q23_webinars',
                     df_a.columns[92]: 'q23_orgnization',
                     df_a.columns[93]: 'q23_leaders',
                     df_a.columns[94]: 'q23_facilitate_trainings',
                     df_a.columns[95]: 'q23_self_training_encouraged',
                     df_a.columns[96]: 'q23_self_training_expected',
                     df_a.columns[97]: 'q23_purchase_publications',
                     df_a.columns[98]: 'q23_others',

                     df_a.columns[99]: 'q24_collaboration_aids', # Q24
                     df_a.columns[100]: 'q25_refer_to_standards', # Q25

                     df_a.columns[101]: 'q26_for_learners', # Q26
                     df_a.columns[102]: 'q26_for_school_librarians',
                     df_a.columns[103]: 'q26_for_school_libraries',

                     df_a.columns[104]: 'q27_inquire', # Q27
                     df_a.columns[105]: 'q27_include',
                     df_a.columns[106]: 'q27_collaborate',
                     df_a.columns[107]: 'q27_curate',
                     df_a.columns[108]: 'q27_explore',
                     df_a.columns[109]: 'q27_engage',

                     df_a.columns[110]: 'q28_think', # Q28
                     df_a.columns[111]: 'q28_create',
                     df_a.columns[112]: 'q28_share',
                     df_a.columns[113]: 'q28_grow',

                     df_a.columns[114]: 'q29_rate_support_materials', # Q29
                     df_a.columns[115]: 'q30_poor_support_reason', # Q30
                     df_a.columns[116]: 'q31_additional_support'} # Q31
                     , inplace=True)

In [16]:
df_a = df_a.iloc[1:].reset_index(drop=True)

In [17]:
df_a.head()

q1_none                   q1_building q1_district_technology  \
0     NaN                           NaN                    NaN   
1     NaN                           NaN                    NaN   
2     NaN                           NaN                    NaN   
3     NaN                           NaN                    NaN   
4     NaN  Building Level Administrator                    NaN   

      q1_district_administrator q1_superintendent q1_school_board q5_none  \
0  District Level Administrator               NaN             NaN     NaN   
1  District Level Administrator               NaN             NaN     NaN   
2  District Level Administrator               NaN             NaN     NaN   
3  District Level Administrator               NaN             NaN     NaN   
4                           NaN               NaN             NaN     NaN   

                            q5_certified  \
0  Certified / licensed school librarian   
1  Certified / licensed school librarian   
2  Certified / licensed school librarian   
3  Certified / licensed school librarian   
4                                    NaN   

                                 q5_noninstructional q5_special_teacher  ...  \
0  Paraprofessional or other non-instructional staff                NaN  ...   
1                                                NaN                NaN  ...   
2  Paraprofessional or other non-instructional staff                NaN  ...   
3                                                NaN                NaN  ...   
4                                                NaN                NaN  ...   

  q27_curate q27_explore q27_engage q28_think q28_create q28_share q28_grow  \
0        NaN         NaN        NaN       NaN        NaN       NaN      NaN   
1        NaN         NaN        NaN       NaN        NaN       NaN      NaN   
2        NaN         NaN        NaN       NaN        NaN       NaN      NaN   
3        NaN         NaN        NaN       NaN        NaN       NaN      NaN   
4        NaN         NaN        NaN       NaN        NaN       NaN      NaN   

  q29_rate_support_materials q30_poor_support_reason q31_additional_support  
0                        NaN                     NaN                    NaN  
1                        NaN                     NaN                    NaN  
2                        NaN                     NaN                    NaN  
3                        NaN                     NaN                    NaN  
4                        NaN                     NaN                    NaN  

[5 rows x 117 columns]

In [18]:
# Drop the first column
df_a = df_a.iloc[:, 1:]

# Drop rows with all NaN values
df_a = df_a.dropna(how='all')

In [19]:
df_a.shape

(30, 116)

In [20]:
# Q17 - Replace extreme values with corresponding integers
df_a_q17 = df_a.iloc[:, [0] + list(range(43, 53))].dropna(
    subset=df_a.columns[43:53], how='all')

df_a_q17.iloc[:, 1:11] = df_a_q17.iloc[:, 1:11].replace({
        'Describes completely': 6,
        '2': 5,
        '3': 4,
        '4': 3,
        '5': 2,
        'Does not describe at all': 0
    }).astype(int)

In [21]:
# Q18 - Two-extremes rescale
df_a_q18 = df_a.iloc[:, [0] + list(range(53, 56))].dropna(
    subset=df_a.columns[53:56], how='all')

df_a_q18.iloc[:, 1:4] = df_a_q18.iloc[:, 1:4].replace({
    "Strongly agree": 2,
    "Somewhat agree": 1,
    "Somewhat disagree": -1,
    "Strongly disagree": -2
}).astype(int)

In [22]:
# Q19 - Two-extremes rescale
df_a_q19 = df_a.iloc[:, [0] + list(range(56, 62))].dropna(
    subset=df_a.columns[56:62], how='all')

df_a_q19.iloc[:, 1:7] = df_a_q19.iloc[:, 1:7].replace({
    "Strongly agree": 2,
    "Somewhat agree": 1,
    "Somewhat disagree": -1,
    "Strongly disagree": -2
}).astype(int)

In [23]:
# Q26 - Four levels
df_a_q26 = df_a.iloc[:, [0] + list(range(100, 103))].dropna(
    subset=df_a.columns[100:103], how='all')

df_a_q26.iloc[:, 1:4] = df_a_q26.iloc[:, 1:4].replace({
    "All the time": 3,
    "Sometimes": 2,
    "Rarely": 1,
    "Never": 0,
}).astype(int)

In [24]:
# Q27 - Four levels
df_a_q27 = df_a.iloc[:, [0] + list(range(103, 109))].dropna(
    subset=df_a.columns[103:109], how='all')

df_a_q27.iloc[:, 1:7] = df_a_q27.iloc[:, 1:7].replace({
    "All the time": 3,
    "Sometimes": 2,
    "Rarely": 1,
    "Never": 0,
}).astype(int)

In [25]:
# Q28 - Four levels
df_a_q28 = df_a.iloc[:, [0] + list(range(109, 113))].dropna(
    subset=df_a.columns[109:113], how='all')

df_a_q28.iloc[:, 1:5] = df_a_q28.iloc[:, 1:5].replace({
    "All the time": 3,
    "Sometimes": 2,
    "Rarely": 1,
    "Never": 0,
}).astype(int)

In [26]:
# OneHot encode
df_a.iloc[:, 0:5] = df_a.iloc[:, 0:5].notna().astype(int) # Q1
df_a.iloc[:, 5:11] = df_a.iloc[:, 5:11].notna().astype(int) # Q5
df_a.iloc[:, 16:25] = df_a.iloc[:, 16:25].notna().astype(int) # Q10
df_a.iloc[:, 26:31] = df_a.iloc[:, 26:31].notna().astype(int) # Q11
df_a.iloc[:, 32:37] = df_a.iloc[:, 32:37].notna().astype(int) # Q12
df_a.iloc[:, 63:78] = df_a.iloc[:, 63:78].notna().astype(int) # Q21
df_a.iloc[:, 86:97] = df_a.iloc[:, 86:97].notna().astype(int) # Q23

## Analysis

`q1`: Which of the following positions most closely reflects your job role? Please select all that apply.

In [27]:
q1_distribution = df_a.iloc[:, 0:5].sum()
print(q1_distribution)

q1_building                  13
q1_district_technology        1
q1_district_administrator    16
q1_superintendent             0
q1_school_board               0
dtype: object


In [28]:
# Calculate the percentage
q1_percentage = (q1_distribution / q1_distribution.sum()) * 100
print("Percentage:\n", q1_percentage)

Percentage:
 q1_building                  43.333333
q1_district_technology        3.333333
q1_district_administrator    53.333333
q1_superintendent                  0.0
q1_school_board                    0.0
dtype: object


In [29]:
# Calculate the number of multiple choices
multiple_choices = (df_a.iloc[:, 0:5].sum(axis=1) > 1).sum()
print("Number of multiple choices:", multiple_choices)

# Filter rows where multiple choices were made
multiple_choices_rows = df_a[df_a.iloc[:, 0:5].sum(axis=1) > 1]
print("Choices for those selecting multiple options:")
print(multiple_choices_rows.iloc[:, 0:5])

Number of multiple choices: 0
Choices for those selecting multiple options:
Empty DataFrame
Columns: [q1_building, q1_district_technology, q1_district_administrator, q1_superintendent, q1_school_board]
Index: []


- There are 30 respondents at all.

- There are only 3 different kinds of administrators, and none of them replied multiple options.

### Utilization

`q11`: Here is an image of the National School Library Standards for Learners, School Librarians, and School Libraries [book] produced by the American Association of School Librarians (AASL)? Please indicate whether you have seen this document before today or have a copy of this document. Please select all that apply.

In [30]:
df_a_q11 = df_a.iloc[:, 26:31]
counts = df_a_q11.sum()
percentages = (counts / 18) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q11_yes_from_workplace     6  33.333333
q11_yes_purchase           5  27.777778
q11_yes_not_own            4  22.222222
q11_yes_from_librarian     3  16.666667
q11_no                     1   5.555556

- The majority of administrators are familiar with the AASL Standards, primarily through workplace-provided copies (33%) or personal purchases (28%).

- A notable proportion (17%) received their copies directly from librarians, highlighting librarians' active role in promoting the standards.

- Only a small minority (6%) have never seen the standards, suggesting strong overall penetration among administrators.

`q12`: Here is an image of the AASL Standards Framework for Learners [pamphlet] produced by the American Association of School Librarians (AASL)? Please indicate whether you have seen this document before today or have a copy of this document. Please select all that apply.

In [31]:
df_a_q12 = df_a.iloc[:, 32:37]
counts = df_a_q12.sum()
percentages = (counts / 18) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q12_yes_not_own            7  38.888889
q12_yes_purchase           5  27.777778
q12_no                     3  16.666667
q12_yes_from_workplace     2  11.111111
q12_yes_from_librarian     2  11.111111

- Most administrators (39%) have seen the pamphlet but don't own a copy, indicating familiarity without personal access.

- Around 28% have purchased their own copies, suggesting strong personal initiative.

- A notable proportion (17%) have never encountered this pamphlet, highlighting potential gaps in distribution or awareness efforts.

`q13`: Did your school or district provide school library personnel with a copy of the National School Library Standards for Learners, School Librarians, and School Libraries book?

In [32]:
value_counts = df_a.iloc[:, 38].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q13_provide_copy                         
Yes                         8   44.444444
Not sure                    5   27.777778
No                          3   16.666667
Other (please specify)      2   11.111111

In [33]:
df_a.iloc[:, 39].dropna().tolist()

['Several copies were purchased to circulate across the district, but indivdual copies were not purchased for each librarian.',
 'Only for library leaders due to the high financial cost']

- Nearly half (44%) confirmed their district provided copies of the standards book to library personnel.

- About 28% of administrators are unsure, indicating potential gaps in communication or awareness.

- Open-ended responses reveal selective purchasing due to budget constraints, with copies sometimes limited to district-wide circulation or only available to library leaders.

`q14`: How familiar are you with the content of the National School Library Standards for Learners, School Librarians, and School Libraries?

In [34]:
value_counts = df_a.iloc[:, 40].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q14_how_familiar                      
Somewhat familiar        8   44.444444
Very familiar            6   33.333333
Not too familiar         3   16.666667
Not at all familiar      1    5.555556

- Most administrators (78%) report being 'Somewhat' or 'Very familiar' with the content of the standards.

- About 22% have limited or no familiarity, suggesting opportunities for further awareness or training initiatives.

`q20`: Does the school librarian use the National School Library Standards for Learners, School Librarians, and School Libraries in their work?

In [35]:
value_counts = df_a.iloc[:, 62].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q20_librarian_using                   
Yes                     10   58.823529
Not sure                 6   35.294118
No                       1    5.882353

- A majority of administrators (59%) confirm that school librarians use the AASL Standards in their work.

- Over a third (35%) are unsure, indicating opportunities for better communication about librarians' implementation of these standards.

`q21`: In which of the following ways have you worked with the school librarian to implement the National School Library Standards for Learners, School Librarians, and School Libraries? Please select all that apply.

In [36]:
df_a_q21 = df_a.iloc[:, 63:78]
counts = df_a_q21.sum()
percentages = (counts / 10) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q21_lesson_preparation_and_assessment              7       70.0
q21_advocate_for_school_library                    7       70.0
q21_learning_opportunities_for_educators           6       60.0
q21_increase_understanding_of_school_librarian     6       60.0
q21_evaluate_librarians_or_libraries               4       40.0
q21_collaboration_with_teacher                     4       40.0
q21_with_other_educators                           4       40.0
q21_guide_instructional_priorities                 4       40.0
q21_with_parents                                   3       30.0
q21_develop_curriculum                             3       30.0
q21_advocate_for_learners                          3       30.0
q21_support_school_mission                         3       30.0
q21_with_external_stakeholders                     2       20.0
q21_demonstrate_learner_success                    1       10.0
q21_none                                           0        0.0

In [37]:
df_a.iloc[:, 78].dropna().tolist()

['educating librarians about their roles',
 'Librarians are involved in everything']

- Most administrators collaborate in lesson preparation and learner assessment (70%) and advocate for the school library (70%).

- Many (60%) partner with librarians to deliver professional learning opportunities or increase awareness about librarians' roles.

- Collaboration extends to instructional priorities, evaluation of libraries, and engaging educators across subjects (40% each), demonstrating diverse integration of standards into school functions.

`q25`: How frequently do you use or refer to the National School Library Standards when working with your school librarian?

In [38]:
value_counts = df_a.iloc[:, 99].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q25_refer_to_standards                   
Rarely                      4        40.0
Sometimes                   3        30.0
Never                       2        20.0
All the time                1        10.0

- Only a small percentage (10%) of administrators frequently reference the standards.

- About 30% sometimes refer to the standards in collaboration with librarians.

- A significant portion rarely (40%) or never (20%) directly use the standards, highlighting opportunities for more consistent implementation.

`q26`: How often do you reference each of the following AASL Standards Frameworks when collaborating with your school librarian?

In [39]:
for col in df_a_q26.columns[1:]:
    print(df_a_q26[col].value_counts().sort_index())

q26_for_learners
2    1
3    2
Name: count, dtype: int64
q26_for_school_librarians
2    2
3    1
Name: count, dtype: int64
q26_for_school_libraries
2    2
3    1
Name: count, dtype: int64


- Administrators most frequently reference the standards framework for learners, with the majority (67%) using it consistently.

- Standards for school librarians are referenced regularly by fewer administrators (33%), but occasionally by most (67%).

- Similarly, administrators occasionally (67%) rather than regularly (33%) reference standards specifically for school libraries.

`q27`: How often do you consider each of the AASL Standards Shared Foundations when collaborating with your school librarian?

In [40]:
df_a_q27.iloc[:, 1:].astype(float).mean()

q27_inquire        3.0
q27_include        3.0
q27_collaborate    3.0
q27_curate         3.0
q27_explore        3.0
q27_engage         3.0
dtype: float64

In [41]:
counts = df_a_q27.iloc[:, 1:].apply(lambda col: col.value_counts().reindex([3, 0], fill_value=0))
print(counts)

   q27_inquire  q27_include  q27_collaborate  q27_curate  q27_explore  \
3            3            3                3           3            3   
0            0            0                0           0            0   

   q27_engage  
3           3  
0           0  


- Among the few administrators (3 respondents) who answered this question, all consistently reference every Shared Foundation when collaborating with school librarians.

- However, the low response count indicates that referencing these Shared Foundations may not be widespread among administrators overall.

### Strength

`q15`: Overall, how would you rate the AASL National School Library Standards?

In [42]:
value_counts = df_a.iloc[:, 41].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q15_rate                    
Good           6   42.857143
Excellent      4   28.571429
Fair           4   28.571429

- The majority of administrators rated the standards as either Good (43%) or Excellent (29%), showing general approval.

- Another 29% rated them as Fair, while none rated them poor or selected Not sure, indicating no strong negative perceptions.

`q17`: Below is a list of words and phrases. Please indicate how well you think each word or phrase describes the National School Library Standards for Learners, School Librarians, and School Libraries.

In [43]:
df_a_q17.iloc[:, 1:].astype(float).mean()

q17_valuable              4.500000
q17_relevant              4.571429
q17_practical             3.928571
q17_useful                4.214286
q17_easy_to_use           4.142857
q17_easy_to_understand    4.142857
q17_well_written          4.500000
q17_nicely_designed       4.428571
q17_well_organized        4.071429
q17_ADA_accessible        4.142857
dtype: float64

In [44]:
counts = df_a_q17.iloc[:, 1:].apply(lambda col: col.value_counts().reindex([6, 0], fill_value=0))
print(counts)

   q17_valuable  q17_relevant  q17_practical  q17_useful  q17_easy_to_use  \
6             4             4              3           3                3   
0             0             0              1           0                1   

   q17_easy_to_understand  q17_well_written  q17_nicely_designed  \
6                       3                 5                    5   
0                       1                 0                    0   

   q17_well_organized  q17_ADA_accessible  
6                   4                   2  
0                   1                   0  


- 'Relevant' received the highest average score (4.57), followed closely by 'Well written' and 'Valuable' (4.50 each), with 4 administrators giving the top rating of 6 for all three terms. 'Nicely designed' also stood out (4.43 average, 5 top ratings).

- While terms like 'Useful', 'Easy to use', and 'Easy to understand' received solid average ratings (~4.14–4.21), each had one respondent who rated them a 0, indicating occasional accessibility or clarity issues.

- 'Practical' had the lowest mean (3.93) and the only instance with both a top (3 responses) and a bottom score (1 response), signaling a more divided view on how actionable the standards are in real-world settings.

`q28`: How often do you consider each of the AASL Standards Domains when collaborating with your school librarian?

In [45]:
df_a_q28.iloc[:, 1:].astype(float).mean()

q28_think     3.0
q28_create    3.0
q28_share     3.0
q28_grow      3.0
dtype: float64

In [46]:
counts = df_a_q28.iloc[:, 1:].apply(lambda col: col.value_counts().reindex([3, 2, 0], fill_value=0))
print(counts)

   q28_think  q28_create  q28_share  q28_grow
3          3           3          3         3
2          0           0          0         0
0          0           0          0         0


- All three administrators who answered this question reported considering each AASL Domain (Think, Create, Share, Grow) all the time when collaborating with school librarians.

- Despite this unanimous enthusiasm, the small number of responses (only 3) suggests that regular use of these domains may not yet be widespread among the broader administrator group.

`q29`: Overall, how would you rate AASL’s efforts to set and maintain up-to-date and practical standards and supporting resources?

In [47]:
value_counts = df_a.iloc[:, 113].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q29_rate_support_materials                   
Good                            7        87.5
Excellent                       1        12.5

- All respondents rated AASL’s efforts as either Good (88%) or Excellent (13%), showing high confidence in the organization’s work.

- No administrators rated the efforts as Fair, Poor, or selected Not sure, indicating a solid and unified endorsement.

### Challenge

`q7`: What are the main reasons you do not work with or collaborate more often with the school librarian?

In [48]:
df_a_q7 = df_a.iloc[:, 12].dropna().tolist()
df_a_q7

['I stay in communication with the librarians as needed, but as a district level supervisor I am tasked with numerous other responsibilities that dominate my work days.',
 'My role is to manage the affairs of all special education students and staff. Our paths just don’t cross as often as I’d like but I fully support our library program. ',
 'The library is housed on another campus']

- District-level duties often take priority, limiting time for librarian engagement.

- One mentioned their primary focus on special education, which naturally reduces overlap with library functions despite their support.

- The logistical challenges due to the library is located on a different campus, making collaboration more difficult.

`q16`: For what reason did you give that rating? Please explain your thinking.

In [49]:
df_a_q16 = df_a.iloc[:, 42].dropna().tolist()
df_a_q16

['They seem very traditional and not forward thinking',
 'The standards at times are overly broad, jargon-y, and hard to understand. Also, the exorbitant cost of the standards book makes it impossible for us to provide access to all of our librarians.  ',
 'It is just too complicated.  Too much.',
 "For library to operate as any other curriculum does, it would be highly beneficial to have grade or age level outcomes for each overall standard; identifying how a first grade student meets proficiency for Inquiry is different than a high school student, but we don't have language available to help break that down. Scaffolded language would be a gift."]

- Administrators described the standards as overly broad, complicated, and difficult to understand, making them challenging to implement effectively.

- High costs and lack of scaffolded, grade-level language were seen as major obstacles to equitable access and practical use.

- One respondent felt the standards were too traditional and not aligned with modern educational needs.

### Suggestion

`q24`: What are some things that could be done to better aid your collaboration with your school librarian to implement the National School Library Standards?

In [50]:
df_a_q24 = df_a.iloc[:, 98].dropna().tolist()
df_a_q24

['More days for her to work in the summer. This is a local contractual issue though. ',
 'Put them in a format that is printable, and easy to refer to.  Provide crosswalks with other national standards to make them easier for librarians to implement.',
 'Create standards that are detailed, specific, and descriptive at different grade levels.',
 'Pare them down.  The book is too long.  ',
 'More cross-walks with other standards (ELA, Math, Science, SS) like there is with ISTE.']

- Administrators called for a shorter, more printable format with clearer language, easier reference tools, and better alignment with other national standards (e.g., ELA, Math, ISTE). These would reduce barriers to practical application.

- Several responses emphasized the need for standards that are detailed and descriptive across grade levels to improve instructional alignment and meaningful implementation.

- One suggestion focused on increasing summer work days for librarians—highlighting how contractual and logistical limitations also hinder collaboration, beyond the content of the standards themselves.

`q31`: What additional supports would have been helpful to you when you first encountered the National School Library Standards?

In [51]:
df_a_q31 = df_a.iloc[:, 115].dropna().tolist()
df_a_q31

['Crosswalks and ties to content standards',
 'Creating a book that was more affordable or putting more things online that can be printed out - I could only find the learner standards in a printable format online',
 'Affordable access to the full standards.',
 'Emphasis on the role of school librarian as a leader and teacher so it would have been more obvious to other educators.']

- Several administrators emphasized the importance of having the full standards available in a more affordable and printable format, beyond just the learner portion. This reflects a broader concern about financial and logistical accessibility.

- Respondents requested crosswalks linking AASL standards to core content areas, suggesting that clearer alignment could aid in adoption and application across disciplines.

- One administrator wished there had been a stronger emphasis on the librarian’s leadership and teaching roles, implying that early framing of the standards could do more to elevate the school librarian’s visibility and influence within the education team.

### With Librarians

`q5`: What does school library staffing look like in your place of work or the school(s) you support? Please select all that apply.

In [52]:
df_a_q5 = df_a.iloc[:, 5:11]
counts = df_a_q5.sum()
percentages = (counts / 26) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q5_certified           24  92.307692
q5_noninstructional    11  42.307692
q5_none                 2   7.692308
q5_volunteer            2   7.692308
q5_special_teacher      0        0.0
q5_not_sure             0        0.0

- A strong majority (92%) of respondents reported having certified or licensed school librarians in their school(s), indicating professional qualifications are the norm in most settings.

- Nearly half (42%) also noted the presence of paraprofessionals or other non-instructional staff, suggesting that while certified librarians lead, additional support roles are common.

- A few schools rely on volunteers (8%) or respondents selected 'None of the above' (8%), pointing to exceptions that may reflect limited funding or alternative staffing models.

`q6`: How often do you collaborate and work with the school librarian(s)?

In [53]:
value_counts = df_a.iloc[:, 11].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q6_how_often                              
On a daily basis            11   52.380952
A couple times per year      4   19.047619
On a weekly basis            3   14.285714
Rarely, if ever              2    9.523810
On a monthly basis           1    4.761905

- A majority (52%) of librarians report working with others every day, highlighting strong integration into school activities.

- About 38% collaborate on a weekly, monthly, or occasional yearly basis, showing moderate but less consistent involvement.

- Nearly 10% rarely, if ever, collaborate—indicating potential barriers such as scheduling conflicts, staffing limitations, or unclear expectations.

`q8`: Who usually initiates collaborations between yourself and the school librarian?

In [54]:
value_counts = df_a.iloc[:, 13].value_counts()
percentage = (value_counts / value_counts.sum()) * 100
result = pd.DataFrame({'Count': value_counts, 'Percentage': percentage})
result.sort_values(by='Count', ascending=False)

Count  Percentage
q8_initiate                                                       
It varies; sometimes me, sometimes someone else     11   57.894737
I do                                                 7   36.842105
Another Administrator                                1    5.263158

- The majority (58%) say collaboration initiation varies, reflecting a flexible or informal process based on context or availability.

- Over a third (37%) of respondents report that they themselves typically initiate collaboration with the librarian.

- Notably, no respondents indicated that their school librarian usually initiates collaboration, which may suggest hierarchical dynamics or uncertainty about the librarian’s role in initiating partnerships.

`q9`: In your opinion, what are the school librarian’s most important contributions to the education program of the school?

In [55]:
df_a_q9 = df_a.iloc[:, 15].dropna().tolist()
df_a_q9

['leader on the campus, collaborator with teachers, professional learning support, literacy support for students and teachers, technology integrator',
 'Elevating and adding value to classroom instruction',
 'Build community, promote literacy and support content area curriculum.',
 'The librarians provide a bridge to students and faculty to connect the classroom curriculum to personal interests, responsible use of resources, and access to current, accurate information.',
 'Providing support and resources to students and teachers. ',
 'Lifting learning outside the walls of the classroom; enhancing lessons through collaborative planning with ALL content area teachers.',
 'The library is the literacy and technology center for the school. ',
 'Education, research, STEM, technology support',
 'It should be supporting classroom teachers and teaching standards in a different way.',
 'Supporting school-level goals; supporting culture and climate; teaching digital citizenship, safety, and liter

- Literacy Champions: Librarians are seen as essential in fostering reading for both pleasure and academic growth, often serving as the literacy hub of the school.

- Guides for Information Navigation: Their role in teaching research skills, information literacy, and digital citizenship highlights their importance in preparing students for an information-rich world.

- Collaborative Instructional Partners: Many responses emphasized collaboration with teachers across all subject areas, enhancing instruction and supporting curriculum goals.

- Equity and Inclusion Advocates: Librarians are valued for curating diverse, inclusive resources and ensuring equitable access for all students, helping foster a supportive and culturally responsive environment.

- Instructional Leaders & Innovators: They are recognized as technology integrators, school leaders, and creative contributors who extend learning beyond the classroom and personalize student support.

`q10`: Which, if any, of the following leadership opportunities in your organization are school librarians involved? Please select all that apply.

In [56]:
df_a_q10 = df_a.iloc[:, 16:25]
counts = df_a_q10.sum()
percentages = (counts / 19) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q10_technology_committee     13  68.421053
q10_professional_learning    13  68.421053
q10_building_leadership      11  57.894737
q10_school_improvement        8  42.105263
q10_curriculum_committee      8  42.105263
q10_district_strategy         6  31.578947
q10_district_leadership       5  26.315789
q10_district_improvement      5  26.315789
q10_none                      2  10.526316

- Librarians are most often involved in technology committees and professional learning (both 68%), showing their key role in digital integration and staff development.

- Over half (58%) participate in building-level leadership, with 42% involved in school improvement and curriculum committees, suggesting active engagement in instructional planning.

- About a quarter of respondents noted librarian participation in district-level leadership, district improvement, and strategic planning, indicating growing but still modest influence at higher administrative levels.

`q19`: Please state how much you agree or disagree with each of the following Common Beliefs of the school library profession.

In [57]:
df_a_q19.iloc[:, 1:].astype(float).mean()

q19_unique_and_essential    1.941176
q19_effective               1.882353
q19_prepared                1.823529
q19_reading                 1.882353
q19_intellectual_freedom    1.882353
q19_equitably_available     1.882353
dtype: float64

In [58]:
counts = df_a_q19.iloc[:, 1:].apply(lambda col: col.value_counts().reindex([2, 1], fill_value=0))
print(counts)

   q19_unique_and_essential  q19_effective  q19_prepared  q19_reading  \
2                        16             15            14           15   
1                         1              2             3            2   

   q19_intellectual_freedom  q19_equitably_available  
2                        15                       15  
1                         2                        2  


- Most respondents strongly agree with each belief, particularly that the school library is essential, reading is core, and intellectual freedom is a right—each endorsed strongly by 15 or more out of 17 participants.

- There were no respondents who disagreed with any belief, reflecting a shared professional identity and consistent values across the group.

- A few respondents selected 'Somewhat agree', particularly on the belief that learners should be prepared for college, career, and life—possibly signaling varied views on how well school libraries currently support this goal.

`q23`: How have you supported professional development opportunities on the National School Library Standards for school librarians in your school or district? Please select all that apply.

In [59]:
df_a_q23 = df_a.iloc[:, 86:97]
counts = df_a_q23.sum()
percentages = (counts / 10) * 100

result = pd.DataFrame({
    'Count': counts,
    'Percentage': percentages
})

result.sort_values(by='Count', ascending=False, inplace=True)
result

Count Percentage
q23_conferences                   8       80.0
q23_leaders                       8       80.0
q23_professional_associations     7       70.0
q23_orgnization                   6       60.0
q23_purchase_publications         6       60.0
q23_workshops                     5       50.0
q23_webinars                      5       50.0
q23_facilitate_trainings          5       50.0
q23_self_training_encouraged      4       40.0
q23_self_training_expected        1       10.0
q23_none                          0        0.0

- The most common supports were conference attendance and district-led training by library leaders (both 80%), followed closely by association memberships (70%) and district-organized training by educational organizations (60%).

- Half of the respondents supported workshops, webinars, and even facilitated training for others, indicating a strong commitment to diverse professional learning formats.

- While 60% supported access to professional publications, fewer encouraged self-directed learning—only 40% did so with time allotted, and just 10% expected it without specific time.